api/kyverno/v2beta1/spec_types.go

```diff
@@ -237,14 +233,28 @@ func (s *Spec) IsGenerateExisting() bool {
 }
 
 // GetFailurePolicy returns the failure policy to be applied
-func (s *Spec) GetFailurePolicy() kyvernov1.FailurePolicyType {
-	if s.FailurePolicy == nil {
-		return kyvernov1.Fail
+func (s *Spec) GetFailurePolicy(ctx context.Context) kyvernov1.FailurePolicyType {
```

Comment:
Can we avoid duplicating this logic ?

Reply:
One possible solution is to add this function for the `WebhookConfiguration` struct instead of `spec`. But, we still need to fetch the `failurePolicy` under the `spec` level if it exists so we still need this function.